books/free-programming-books-subjects.md

```diff
@@ -828,7 +828,11 @@ Kerridge (PDF) (email address *requested*, not required)
 
 * [DALLE-E 2 prompt book](https://dallery.gallery/wp-content/uploads/2022/07/The-DALL%C2%B7E-2-prompt-book.pdf) - Dallery.Gallery, Guy Parson (PDF)
 * [Guide to Prompt Engineering](https://codeahoy.com/learn/promptengineering/toc) - CodeAhoy (HTML)
+* [LangChain: Introduction and Getting Started](https://www.pinecone.io/learn/series/langchain/langchain-intro/) - Pinecone Team (HTML)
+* [LangGraph Cookbook](https://github.com/abhishekmaroon5/langgraph-cookbook) - Abhishek Maroon (GitHub)
 * [LLM Prompt Engineering Simplified Book](https://github.com/AkmmusAI/LLM-Prompt-Engineering-Simplified-Book) - Katikapalli Subramanyam Kalyan (GitHub)
+* [Model Context Protocol (MCP) - Complete Deep Dive](https://www.scribd.com/document/880507979/Model-Context-Protocol-MCP-Complete-Deep-Dive) - Author (PDF)
+* [Model Context Protocol (MCP) - Guide With Demo Project](https://www.scribd.com/document/885885278/Model-Context-Protocol-MCP-a-Guide-With-Demo-Project-DataCamp) - DataCamp Team (PDF)
```

Comment:
This appears not to be an authorized distribution. Please see docs/CONTRIBUTING.md guidelines.